### PR TITLE
implementation of multiple proving sys type rebased on sidechains_dev

### PR DESCRIPTION
**Pull request by alsala**

### CODE REVIEW

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -1280,7 +1280,7 @@ UniValue createrawcertificate(const UniValue& params, bool fHelp)
 
             std::string error;
             std::vector<unsigned char> cmt;
-            if (!Sidechain::AddScData(o.get_str(), cmt, MAX_CMT_SIZE_BYTES, false, error))
+            if (!Sidechain::AddScData(o.get_str(), cmt, MAX_CMT_SIZE_BYTES, Sidechain::CheckSizeMode::UPPER_LIMIT, error))
```
Comment (albertog78) [inline src/rpc/rawtransaction.cpp:1283]: MAX_CMT_SIZE_BYTES should be BitVectorCertificateFieldConfig::MAX_COMPRESSED_SIZE_BYTES

**Review by albertog78 — CHANGES_REQUESTED** [review]

File: src/coins.cpp [hunk 2]
```diff
@@ -1565,7 +1583,12 @@ CValidationState::Code CCoinsViewCache::IsScTxCswProofVerified(const CTransactio
         // TODO note that actCertDataHash can be null: in this case a Phantom hash shall be passed to the verifier
 
         // Verify CSW proof // TODO update this call as soon as the final signature is defined
+#if 0
         if (!scVerifier.verifyCTxCeasedSidechainWithdrawalInput(certDataHash, sidechain.fixedParams.wCeasedVk.get(), csw))
+#else
+        scVerifier.LoadDataForCswVerification(*this, tx);
```
[the remark this answers is not in the document]
Comment (alsala) [reply src/coins.cpp:1543]: This implementation will be superseded by branch pt/new_mempool_batch_verification

File: src/sc/proofverifier.cpp [hunk 3]
```diff
@@ -1,172 +1,174 @@
...
+    }
 
-    bool CScWCertProofVerification::verifyScCert(                
-        const ScConstant& constant,
-        const ScVk& wCertVk,
-        const uint256& prev_end_epoch_block_hash,
-        const CScCertificate& scCert) const
-    {
-        // Collect verifier inputs
+    certData.quality = scCert.quality; //Currently quality not yet accounted for in proof verifier
```
[the remark this answers is not in the document]
Comment (alsala) [reply src/sc/proofverifier.cpp:49]: This implementation will be superseded by branch pt/new_mempool_batch_verification

File: src/sc/proofverifier.cpp [hunk 4]
```diff
@@ -1,172 +1,174 @@
...
+        if (sidechain.fixedParams.wCeasedVk.is_initialized())
+            cswData.ceasedVk = sidechain.fixedParams.wCeasedVk.get();
+        else
+            cswData.ceasedVk = CScVKey{};
 
-            LogPrint("zendoo_mc_cryptolib",
-                "%s():%d - failed to deserialize \"wCertVk\": %s \n", 
-                __func__, __LINE__, ToString(zendoo_get_last_error()));
-            zendoo_clear_error();
+        cswData.cswOut = csw;
```
[the remark this answers is not in the document]
Comment (alsala) [reply src/sc/proofverifier.cpp:98]: This implementation will be superseded by branch pt/new_mempool_batch_verification

File: src/sc/proofverifier.cpp [hunk 5]
```diff
@@ -1,172 +1,174 @@
...
+    return res;
+}
 
-        if(inputs.sc_proof == nullptr) {
+#ifdef BITCOIN_TX
+void CScProofVerifier::LoadDataForCswVerification(const CCoinsViewCache& view, const CTransaction& scTx) {return;}
+#else
+void CScProofVerifier::LoadDataForCswVerification(const CCoinsViewCache& view, const CTransaction& scTx)
+{
+    for(size_t idx = 0; idx < scTx.GetVcswCcIn().size(); ++idx)
```
[the remark this answers is not in the document]
Comment (alsala) [reply src/sc/proofverifier.cpp:79]: This implementation will be superseded by branch pt/new_mempool_batch_verification.

File: src/rpc/rawtransaction.cpp (hunk 1 above)
Comment (alsala) [reply src/rpc/rawtransaction.cpp:1282]: Done

**Pull request merged**